### DISCUSSION TITLE
possibilità di gestione/eliminazione dati dell'app (ability to manage/delete app data)

### DISCUSSION THREAD

**Discussion by Ma-80** [osmandapp/OsmAnd#21533] [Q&A]
Buongiorno, scusate ma non parlo e non capisco l'inglese tanto bene da esporre il quesito. Quindi scriverò in italiano.
(Good morning, sorry but I don't speak or understand English well enough to ask the question. So I will write in Italian.)

Sto cercando di fare un po' di spazio nella memoria interna del cellulare e per questo ho spostato l'app nella memoria esterna (scheda SD), ma dallo spazio che si è liberato pare abbia spostato solo l'app e non anche i dati (mi riferisco a quelli ai quali si riferisce la schermata allegata, presa dalle impostazioni del telefono nella sezione gestione applicazioni). Specifico che dalla vs app avevo già impostato il salvataggio nella memoria esterna (cartella /storage/6331-3730/Android/data/net.osmand/files).
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/e94f0375-6600-479a-9f64-19eba3d977a7"  width="300" height="auto"/> </td>
  </tr>
</table>

Vorrei capire se si tratta degli stessi dati, cioè se nei 1,16 GB indicati nella voce dati visualizzati nella gestione applicazioni rientrano anche quelli che vengono salvati nella memoria esterna in base alle impostazioni dell'app come sopra esposto, oppure se si tratta dei soli dati scaricati nella memoria interna e riguardano dati diversi dalle tracce e i dai preferiti salvati.

Posso quindi cancellare i dati da gestione applicazioni senza cancellare le tracce registrate e i preferiti?

Di seguito le informazioni sul dispositivo e sulla versione dell'app:

Device : a3ulte
Brand : samsung
Manufacturer : samsung
Model : SM-A300FU
Product : a3ultexx
Build : MMB29M.A300FUXXU1CRF1
Version : 6.0.1
App Version : OsmAnd
Apk Version : 4.6.11 4611

Spero di aver esposto il quesito in modo chiaro e che mi sappiate dare le informazioni richieste.

Grazie, saluti,
Mary

**Comment by Sonwon1** [comment]
Google Translate is your friend!

**Reply by Ma-80** [reply]
Lo so. Siccome l'applicazione è fornita anche in italiano, ho sperato che l'assistenza e/o gli sviluppatori capissero la mia lingua o avessero degli interpreti. (I know. Since the application is also provided in Italian, I hoped that the support and/or developers would understand my language or have interpreters. _[translated with Google Translate]_)

**Comment by Ma-80** [comment]
I translate for those who do not know Italian (I used Google Translate, so I do not know how reliable the translation is).

I am trying to make some space in the internal memory of the phone and for this I moved the app to the external memory (SD card), but from the space that was freed it seems that I only moved the app and not the data (I am referring to those referred to in the attached screenshot, taken from the phone settings in the application management section). I specify that from your app I had already set the saving in the external memory (folder /storage/6331-3730/Android/data/net.osmand/files).
I would like to understand if it is the same data, that is, if the 1.16 GB indicated in the data item displayed in the application manager also includes those that are saved in the external memory based on the app settings as explained above, or if it is only the data downloaded to the internal memory and concerns data other than the tracks and saved favorites.

Can I therefore delete the data from the application manager without deleting the recorded tracks and favorites?

Below is the information on the device and the app version:
Device : a3ulte
Brand : samsung
Manufacturer : samsung
Model : SM-A300FU
Product : a3ultexx
Build : MMB29M.A300FUXXU1CRF1
Version : 6.0.1
App Version : OsmAnd
Apk Version : 4.6.11 4611

I hope I have explained the question clearly and that you can give me the information requested.

Thanks, regards,
Mary

**Comment by Ma-80** [comment]
A quanto pare il problema non è la lingua, ma chi legge (o forse non legge). Mi spiego:
Il supporto dell'app mi ha indicato questo sito per contattare gli sviluppatori, ma non ho capito se queste discussioni vengono lette dagli sviluppatori oppure se neanche loro sanno cosa e dove viene memorizzato dall'app nella memoria del telefono.
Il quesito mi sembrava abbastanza chiaro. O no? 
O forse non hanno tempo da perdere per queste cose (non mi aspetto chissà cosa avendo una app gratuita).
Grazie comunque
___

Apparently the problem is not the language, but who reads (or maybe doesn't read). Let me explain:
The app support told me this site to contact the developers, but I didn't understand if these discussions are read by the developers or if they don't even know what and where the app stores in the phone's memory.
The question seemed clear enough to me. Or not?
Or maybe they don't have time to waste on these things (I don't expect much having a free app).
Thanks anyway

**Comment by MarioSwitch** [comment]
You can export favorites and tracks in a `.osf` file (in the in-app Settings menu), then delete your data, and import your file again.

**Comment by Ma-80** [comment]
Un dubbio però mi è rimasto.
Mi sembra di capire che cancellando i dati dell'app si eliminano anche tracce e preferiti, quindi a questo punto si aprono due possibilità:
1. Se i dati sono salvati tutti insieme, allora essi sono nella memoria esterna (dove ho impostato di salvare le tracce) e cancellandoli non libero spazio nella memoria interna del telefono (che è il mio obiettivo).
2. Se i dati non sono salvati tutti insieme e la maggior parte sono nella memoria interna del telefono, allora vale la pena provare a cancellarli.

Qualcuno sa quale delle due possibilità è giusta?
Grazie ancora,
Mary
__

However, I still have a doubt.
It seems to me that deleting the app data also deletes tracks and favorites, so at this point there are two possibilities:
1. If the data is saved all together, then they are in the external memory (where I set to save the tracks) and deleting them does not free up space in the internal memory of the phone (which is my goal).
2. If the data is not saved all together and most of it is in the internal memory of the phone, then it is worth trying to delete it.

Does anyone know which of the two possibilities is right?
Thanks again,
Mary

**Reply by MarioSwitch** [reply]
As far as I know, all data used by OsmAnd needs to be in the same folder (answer 1).
You can view which folder is used (and by extension on which memory is it) in `Settings > OsmAnd Settings > Storage folder`.

You can also view how OsmAnd data takes on your storage (and manage it) in `Maps & Resources > Storage`.

**Comment by Ma-80** [comment]
Ora che ci penso, quando ho spostato l'app nella memoria esterna quel giga non è cambiato, quindi ho l'impressione che non abbia spostato tutti i dati, che pertanto devono essere ancora per la maggior parte nella memoria interna (le tracce invece le avevo già impostate per l'archiviazione su scheda sd). 
A questo punto proverò a cancellare i dati e vediamo che succede (tanto ho fatto il backup di tracce, preferiti e impostazioni come suggerito da MarioSwitch).
Le mappe scaricate invece non ce la fa a fare il backup, quindi presumo che dovrò riscaricarle, anche se non ricordo come ho fatto.
Grazie 
___

Now that I think about it, when I moved the app to external storage that gigabyte didn't change, so I have the impression that it didn't move all the data, which must therefore still be mostly in internal storage (I had already set the tracks to be stored on an SD card).
At this point I'll try to delete the data and see what happens (I backed up the tracks, favorites and settings as suggested by MarioSwitch).
The downloaded maps, on the other hand, can't be backed up, so I assume I'll have to re-download them, although I don't remember how I did it.
Thanks

**Reply by MarioSwitch** [reply]
You can download maps in the `Maps & Resources` tab 😉

**Reply by Ma-80** [reply]
In mappe e risorse non capisco dove cliccare.
Sarò rimbambita o cecata...

In maps and resources I don't understand were to click.
I'm either senile or blind...

<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/1e8c9d69-427a-429e-b37d-6faf860999be"  width="300" height="auto"/> </td>
  </tr>
</table>

**Reply by MarioSwitch** [reply]
You should have this:
<table style="width:100%">
  <tr>
<th><img src="https://github.com/user-attachments/assets/5bf06c7d-a619-451b-8fba-a86847194c7e"  width="300" height="auto"/> </td>
  </tr>
</table>

Maybe you're not connected to the internet? You need internet connection to download maps...

**Reply by Ma-80** [reply]
Forse il fatto che io non ho nulla nella sezione download dipende dalla mia versione (forse on aggiornata all'ultima disponibile), ma non fa niente.
Maybe the fact that I have nothing in the download section depends on my version (perhaps not updated to the latest available), but it doesn't matter.

**Comment by Ma-80** [comment]
Esperimento fallito.
A quanto pare i dati non erano salvati nella memoria interna, ma tutti nella scheda sd dove ho spostato la app e ho impostato la cartella di salvataggio. Quando ho cancellato i dati non si è liberato nessuno spazio nella memoria del telefono. Presumo che i dati non siano mai stati nella memoria interna visto che l'impostazione di memorizzazione su sd l'avevo fatta già all'inizio. Ho solo perso tempo inutilmente a scombinare e cercare di riaggiustare le impostazioni dell'app. Per fortuna avevo fatto un backup anche delle impostazioni. 
Morale della favola: con questo tipo di app non cancellare i dati e in generale non pretendere che funzionino a lungo, altrimenti gli sviluppatori rimarrebbero disoccupati.

**Reply by MarioSwitch** [reply]
If your data was already in the SD card, why did you expect to save storage by uninstalling the app/deleting data?

> Morale della favola: con questo tipo di app non cancellare i dati e in generale non pretendere che funzionino a lungo, altrimenti gli sviluppatori rimarrebbero disoccupati.

Please, don't act in bad faith... My app never had problems. Your main subject was "ability to manage/delete app data" which is possible and I've already explained many times how to do. The discussion would be closed now.

**Comment by Ma-80** [comment]
"If your data was already in the SD card, why did you expect to save storage by uninstalling the app/deleting data?"

Hai ragione, non ero sicura che fossero già tutti nella scheda sd. Sono una cretina.
You're right, I wasn't sure if they were all already on the sd card. I'm an idiot.

"Please, don't act in bad faith... "

Non sono io in malafede, quegli sviluppatori che fanno aggiornamenti che con la scusa di migliorare delle cose ne storpiano delle altre. E li rendono pure obbligatori. Non sarà questo il caso, ma per colpa di quella gente ci vanno di mezzo anche le poche persone serie. Ed ora io odio gli aggiornamenti. Quindi scusa se non mi sono fidata al 100% di quello che hai scritto e se mi sono sfogata qui.

Direi che possiamo chiudere la conversazione.

Grazie comunque

I'm not in bad faith, those developers who make updates that with the excuse of improving some things distort others. And they also make them mandatory. This may not be the case, but because of those people even the few serious people suffer. And now I hate updates. So sorry if I didn't trust 100% of what you wrote and if I vented here.

I think we can end the conversation.

Thanks anyway
